### DISCUSSION TITLE
RSA private key

### DISCUSSION THREAD

**Discussion by filipgrzelak** [kjur/jsrsasign#593] [Q&A]
i want to sign a data with KJUR library.

I have got RSA PRIVATE KEY which looks like that.

`const PRIVATE_KEY = "\---- BEGIN RSA PRIVATE KEY---- ... ----END RSA PRIVATE KEY----
`
and i want to sign data like this

`const keyPEM = KEYUTIL.getKey(PRIVATE_KEY)`

and when i want compile this code this error occurs: "Error: unsupported PKCS#1/5 hexadecimal key"

What should i do to resolve this problem. I have to use KJUR library. Thanks for help

I try a lot of combinations of examples and i dont have more ideas

**Comment by kjur** [comment]
Can you send me your key? Then I can do investigation.

**Reply by filipgrzelak** [reply]
"-----BEGIN RSA PRIVATE KEY-----\n"+
    "MIIEowIBAAKCAQEA8xU31UaOXi63p0rtLoadDB5seQpccIDkx1fws1S9oT2RCKWx\n"+
    "JchKed89/KiG/G78YK5yvX+CgjkmgZuccSUw0BgG/dUo7UhwtuZT1zzI3b8KMUmR\n"+
    "T0olaLFf/kwtiBHQoVlQxusPhinWqDy54gi261b1GLlJubRLYzzIXOQ9YhU3I5bU\n"+
    "gpaGtUyvVy5ZFaJs2J4bZu4i+UTU4MpSgdlvJrxzKhBXELDlo/qMBpgaydjNmj/Y\n"+
    "7qeMagyjD6omBvk8fDvxH9g6+sGM1y1HoQt2DU/lEmUo/o5tVSzh/MQd7IS95lRt\n"+
    "m9pSJSdfjoJjxOQe4gEdNvf6u5ZKRPP3HeJb9QIDAQABAoIBABGAHFAuP3cv4Vh5\n"+
    "uDgyiFF0SkZvlMajvPTOE73xcqKOYmfvN+D6ucPq6tVLE8bOPby/KnMSoER8VVWc\n"+
    "7KcnsIs2w6AYLdVhcv3z+KLFFUgDpjn0rDG7apSOOpU+CkQi1jSbj8l2aFx2HCZq\n"+
    "Es7fZj3suYIRaumj/Y50NI+dff7W2MFN7jXw0p0K0edHQ1J4EOOQMc6lolOtPwnR\n"+
    "1CxwLSUfPsLa2HsETEpSRymApP6Xgmr/qb5q3a7cAipDYwkIJ//3gfzupnbhWIMu\n"+
    "tqR2ljMvQu1TEX5ZJCyKCQ3jP/5imqDV0N1UmKYTGmaSlvE2d/zsa6Qwb5auuHs7\n"+
    "Sghg/vECgYEA+XtRBSXMFAkQdhxJMS4CwiFnBBonr+lJ8up04BbgKsR01IL7n8xg\n"+
    "Uap9i65XJXWaQMxt0TN67TZIUVu0ZWThR08+qaukyP2DVPM1eT5KAGL9vAJnexo9\n"+
    "OAS39tSY7EnF3XQe9F1t0spc2xVe0DlXfw1hl/VFio62MZ1/TP32WMUCgYEA+W8a\n"+
    "OhgBEZKriRrtsirM6WO42sjMF/l5CHRMr0szWFyPwuQkT2z3L5QM5JOtFHXC+m49\n"+
    "2ofF5ML0kPPN9zGDPIf3Djcxz0GMwwFEr1PPFTw1J1y9ADD5wfeA/9VJoArOVuE4\n"+
    "+dbql8LeyLE93P5Ekru124Ubj3Y6aZTmCNzRSXECgYEAhcuPsO4LW3dc/6q9ygU9\n"+
    "73TQdY+9VwcK5J6+nWtHP0hqy4mPfqyxvzpq6B1QmUNPvY6lGnu8YjkpYZCH15TZ\n"+
    "iE/EcGQgFLZ413czZ1lzgG+WWl0DuC5jRIarTlHHzXjnB5P+AdmVS2XBlLEKTyth\n"+
    "dD7t5vvgFqXgqg0oN1c/5W0CgYAkq7vcXJkTn0+OuaHGdtpnWZzctXaj865ims6H\n"+
    "WvPygrE0Atx/vNJecVKv7DIe9u0yajC3xwxGBI92HBHkzKexp3+f3Gd7Q/TQ2ofh\n"+
    "it3ADnj2D4FN3OVrKdsSQaYwXEZXqYb53CnjI+gi80PYEJ2+QNCMtHYH0mbmDNkA\n"+
    "5o3yUQKBgDKHYdzOOHB0MN5X5mM3AGC3xfOYQLkiNIfWLyp6SDJbpk7K8RK5Qocp\n"+
    "f2DMNKEEfob80q5Z1dZxe2OOvKgMndGpxlmZ7BFJO9usUqY8vOc+ZEridA4IvVqI\n"+
    "iUfVXbCc4d7tpk+02pZ90jKnqPQ3pIeWOFMGj0hm35pLcps/a6ga\n"+
    "-----END RSA PRIVATE KEY-----\n"

**Reply by filipgrzelak** [reply]
this is my key

**Reply by filipgrzelak** [reply]
and it is generated with commands you sent me below. the error is the same

**Reply by filipgrzelak** [reply]
when i remove `\n` at the end of each line the error still occurs

**Comment by kjur** [comment]
Otherwise, can you use another private key created by following commands:
```
# generating RSA 2048bit PKCS#8 format private key having "-----BEGIN PRIVATE KEY-----"
% openssl genrsa 2048 > a.prv.p8
# convert it from PKCS#8 to PKCS#1/#5 format like yours having "----BEGIN RSA PRIVATE KEY-----" 
% openssl rsa -in a.prv.p8 -out a.prv.p5 -traditional
```
Then could you confirm the keys "a.prv.p8" and "a.prv.p5" works?

**Reply by filipgrzelak** [reply]
ok, finally after a few tries it start works. Thanks
